fix(lib/modules/manager/gitlabci): fixed image regex

hasanwhitesource opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->

## Changes

Fixed regex to skip a string if it ends with `:`

## Context

Closes #14340

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

viceice reviewed (changes requested):
Please add a test.

viceice commented on lib/modules/manager/gitlabci/__fixtures__/gitlab-ci.8.yaml:
```diff
@@ -0,0 +1,33 @@
+job1:
```
Can't we simply add a new value to existing test instead of crfeating this huge new test?

hasanwhitesource commented:
I have removed a few cases and moved the most important check to the `extracts multiple named services` test which I think is the most relevant to this case.

hasanwhitesource commented:
@rarkins any updates here?
